### DISCUSSION TITLE
New Scottish Snowpatches Sentinel-2 Data web tool

### DISCUSSION THREAD

**Discussion by EddieBoyle2019** [SimonFisher92/Scottish_Snow_Satellite_Data#36] [General]
Hi all,
 
Murray and I have collaborated on developing a web tool which is now a functional beta prototype, targeted towards the snowpatch observing community. I will be getting feedback on this from Iain Cameron. 

It uses Streamlit, the SentinelHub API, Folium for mapping, an OSM basemap, the Sentinel 2 RGB imagery layer and the algorithm-generated Scene Classification Map layer to display snow/ice feature classes (selected from the layer selector icon at the top right of the map).

It allows a user to select from a list of targeted snowpatch areas (this list will become more useful soon). A date period is also selected and the least cloudy image is used from that period. Transparency of the layers is also configurable for visualisation purposes.

You will need to enter your CDSE account credentials to use it.

https://gnmkt4fm3cv4tzxtbx54bg.streamlit.app

There are limitations to this of course, notably images with significant cloud cover, and the SCM algorithm misses some snowpatches, but this shows what is possible right now - at some point in the future I envisage the work of this group adding a new layer to this tool with more accurate identification of snow features.

**Comment by ipoole** [comment]
Hi Eddie, Murray.   I've just been looking at this (once I managed to track down my CDSE credentials...).   Yes this is nice.   Some thoughts on enhancements:
* Show the date of the image displayed
* Optionally overlay (checkbox) the SCM snow classification.  This would then provided a useful means of reviewing the SCM labels.
* Show the area (in m^2) of snow within the patch according to SCM.

Well done!

**Comment by EddieBoyle2019** [comment]
Some good thoughts there Ian, thanks. I also had some feedback on the tool from Iain Cameron:

Features to add:
1. A date stamp showing which day was being shown from the date-range would be very useful. It's always good to know what you're looking at.
2. A 'Forward' or 'Back' arrow which takes the viewer to the next available cloud-free date, or previous one, would be useful.
3. If possible, a grid reference would be good to include where the blue pin is.
4. I would prefer to see the blue box which shows the satellite image as large as the map overlay. It just looks a little small to my eyes.

General view of the tool:

"I don't think it'll be used widely among the group as most are just interested observers. However, I can see a hardcore rump using it. Plus I think that students would make use of it. I get contacted a lot on FB and Twitter with questions that something like this would at least go some way to addressing"

**Comment by EddieBoyle2019** [comment]
Hello all,

The web app developed my Murray and myself with input from everyone in the group, is now ready for release.

I will do this sometime this week with a posting to the snowpatches Facebook group.

The URL is:
https://scottish-snowpatches.streamlit.app

Comments welcome!

**Reply by murraycutforth** [reply]
Great job again Eddie, it's great fun to scroll around looking at the hills, and does make me very homesick as well!

I noticed that both Ian and Iain mentioned knowing the date of the displayed image would be helpful. I remember thinking that the way the API is configured made that information suprisingly difficult to extract, but maybe that is the logical next step for the web app and we can make an issue to see if this is possible?

**Reply by EddieBoyle2019** [reply]
Yes, I had a look at this and it's not an easy thing to do (largely because the images are combined via mosaicking pixel-by-pixel over a date range and finding a single cloud-free image on one chosen day is not easy)! I discuss this issue in the help feature of the app, and I direct people to use the Copernicus Browser if they want more control over the images, which I think is the best approach here. This app is more of a simple guide to let people explore the appearance and context of the snowpatches within the wider environment of the Scottish Highlands via satellite imagery and OpenStreetMap web mapping. For serious snowpatch data (i.e. melt dates), the Copernicus Browser is the resource to use.

**Reply by EddieBoyle2019** [reply]
The best way to deal with this issue and use the SentinelHub API is to allow a user to choose a percentage cloud cover value, and then show graphically on a calendar interface those dates with images available for the area that have cloud cover less than the chosen value - which is exactly what the Copernicus Browser does. I don't think we should necessarily be replicating this in our app....but maybe this indeed is something for v2.0....

**Reply by EddieBoyle2019** [reply]
Something else we can add to the list of things for a new version is support for smartphones and small screens, it doesn't really work properly on these.

**Reply by EddieBoyle2019** [reply]
And yes, Simon's Facebook comment raises a valid point, the current system, where users need a personal SentinelHub  OAuth client account to use the tool, means that data quotas are never going to be a problem, but it's still a hurdle for some people. If we can find a better way to do this we should, we can investigate if a group account is possible.
